src/app/styles-registry.tsx

```diff
@@ -0,0 +1,31 @@
+/* istanbul ignore file */
+"use client";
+
+import React, { useState } from "react";
+import { useServerInsertedHTML } from "next/navigation";
+import { ServerStyleSheet, StyleSheetManager } from "styled-components";
+
+// https://nextjs.org/docs/app/building-your-application/styling/css-in-js#styled-components
+export default function StyledComponentsRegistry({
```

Comment:
I noticed this was missing — it's needed to ensure that styled-component styles are server side rendered. Until now there was a noticeable flicker of styles while the CSS loaded.